functions/users.inc

```diff
@@ -99,73 +99,185 @@ function lookup_user_by_id($id)
 }
 
 
-function lookup_users_by_email($email, $single = true)
+function lookup_users_by_email($email, $single = true, $substring = false)
 {
-    return _lookup_user([['Email', 'EQUAL', $email]], $single);
+    if ($substring) {
+        $operator = 'CONTAIN';
+    } else {
+        $operator = 'EQUAL';
+    }
+    return _lookup_user([['Email', $operator, $email]], $single);
 
 }
 
+$_set = array();
 
-function lookup_users_by_name($name, $single = false)
+function filter_users($user)
 {
-    $names = explode(" ", $name);
-    if (count($names) != 2) {
-        return ['code' => '400 Bad Request', 'users' => array()];
+    global $_set;
+    if (in_array($user['Id'], $_set)) {
+        return false;
+    } else {
+        $_set[] = $user['Id'];
+        return true;
     }
-    return _lookup_user(
-        [['First Name', 'EQUAL', $names[0]],
-                         ['Last Name', 'EQUAL', $names[1]]],
-        $single
-    );
-
 }
 
+function sort_users($a, $b)
+{
+    return intval($a['Id']) > intval($b['Id']);
+}
 
-function lookup_users_by_pref_name($name, $single = false)
+function merge_results($result1, $result2)
 {
-    $preferredFirstName = search_definedFields('Preferred First Name');
-    $preferredLastName = search_definedFields('Preferred Last Name');
-    $names = explode(" ", $name);
-    if (count($names) != 2) {
-        return ['code' => '400 Bad Request', 'users' => array()];
+    global $_set;
+    $_set = array();
+    $output = ['code' => '404 Not Found', 'users' => array()];
+
+    if ($result1['code'] == null) {
+        $output['code'] = null;
+        $output['users'] = array_merge($output['users'], $result1['users']);
+        foreach ($result1['users'] as $u) {
+            $_set[] = $u['Id'];
+        }
     }
 
-    /* We have to do a few searchs due to Neon stupidity */
-    $output = _lookup_user(
-        [[$preferredFirstName, 'EQUAL', $names[0]],
-         [$preferredLastName, 'EQUAL', $names[1]]],
-        $single
-    );
-    if (count($output['users']) != 0) {
-        return $output;
+    if ($result2['code'] == null) {
+        if ($output['code'] === null) {
```

Comment:
Why `===` here, but `==` above? I know they're subtly difference; just not clear if the difference matters here.